src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -87,11 +87,11 @@ module.exports = class LevelPlane extends Array {
   * Changes the block at a desired position to the desired block.
   * Important note: This is the cornerstone of block placing/destroying.
   */
-  setBlockAt(position, block) {
+  setBlockAt(position, block, force = false, skipRedstone = false) {
     this[this.coordinatesToIndex(position)] = block;
 
     let redstoneToRefresh = [];
-    if (block.isRedstone || block.blockType === '' || block.isConnectedToRedstone) {
+    if (!skipRedstone && (block.isRedstone || block.blockType === '' || block.isConnectedToRedstone)) {
```

Comment:
Why do these updates skip redstone propagation?  Can pistons move blocks (or entities) onto a piston that triggers a redstone update?

Reply:
I was overflowing the stack because evaluate redstone -> evaluate piston -> pushes blocks with setBlockAt() which -> re-evaluates redstone etc etc.

It's basically a case where I was changing a block that was connected to redstone but I didn't want to re-evaluate all the redstone. I can look into making this cleaner if need be.

Reply:
I'm definitely in favor of making this cleaner; I think we're opening ourselves up to some inconsistent/unpredictable redstone update behavior with this.